### ISSUE TITLE
sceAac needs implementing

### ISSUE THREAD

**Issue by scraggles80** [hrydgard/ppsspp#3870]
games like Knight Fortix 2 (doesn't boot) and Zodiac (freezes up during the intro)
need sceAac added

Games using sceAac:
NPUZ00300 - Knight Fortix 2
NPJH00028 - Zodiac
NPUZ00016 - Fortix
NPJH00092 - Nozomi Kanaetamae - Daydream Reconstruct

Games using sceMp4Aac:
NPUZ00256/NPEZ00390 - Speedball 2 Evolution
NPUH10069 - Auditorium

**Comment by benderscruffy** [comment]
v1.5.4-933-g386c9d4fc
still present  Speedball 2 Evolution nothing but a black screen

**Comment by benderscruffy** [comment]
Speedball 2 Evolution goes ingame
lots of unimplemented scemp4
43:37:865 movieThread  E[HLE]: hle\hle.cpp:545 Unimplemented HLE function sceMp4AacDecodeExit
43:37:865 movieThread  E[HLE]: hle\hle.cpp:545 Unimplemented HLE function sceMp4AacDecodeTermResource
43:37:881 movieThread  E[HLE]: hle\hle.cpp:545 Unimplemented HLE function sceMp4TrackSampleBufDestruct
43:37:881 movieThread  E[HLE]: hle\hle.cpp:545 Unimplemented HLE function sceMp4UnregistTrack
here is a debug log
[ppsspplog.zip](https://github.com/hrydgard/ppsspp/files/4014017/ppsspplog.zip)
![speedball](https://user-images.githubusercontent.com/2696404/71640973-5a1e2980-2ccf-11ea-9975-758503cb5281.PNG)

**Comment by Panderner** [comment]
SceAac uses AAC. AAC files can be played in most devices, Windows, IOS, Android, can play AAC files
I don't think I needed to implement AAC via ffmpeg or faac

**Comment by hrydgard** [comment]
We try to not rely on platform APIs for stuff like that wherever possible, because there's always a chance that they'll behave differently on one platform than another. So it's best to use FFMPEG. Either way, that's not really the problem here, it's just that the sceAac API needs to be better figured out and hooked up.

**Comment by benderscruffy** [comment]
PPSSPP v1.13.1-234-g5434bd83e
Speedball 2 Evolution goes ingame
missing all of the videos
lots of unimplemented sceMp4 and sceMpeg
sceMp4Create
sceMp4AacDecodeInitResource
sceMp4AacDecodeInit
sceMp4RegistTrack
sceMp4TrackSampleBufConstruct
sceMp4InitAu
sceMp4GetMovieInfo
sceMpegAvcResourceGetAvcEsBuf

[ppsspplog.txt](https://github.com/hrydgard/ppsspp/files/9278784/ppsspplog.txt)

**Comment by benderscruffy** [comment]
Nozomi Kanaetamae - Daydream Reconstruct - NPJH00092
music cuts off just in the intro otherwise no problems at all 
I[ME]: HLE\sceMp4.cpp:215 UNIMPL sceAacInit(09fff158)
[ppsspplog.zip](https://github.com/hrydgard/ppsspp/files/9305897/ppsspplog.zip)

**Comment by FATCatAndroid12** [comment]
I tried Knight Fortix 2 no audio issue

![Image](https://github.com/user-attachments/assets/94065072-a721-45df-8ef9-b75d49e46f47)

Speedball Evolution 2 stay on blackscreen but pressing start can start the game similar to this https://github.com/hrydgard/ppsspp/issues/15004

![Image](https://github.com/user-attachments/assets/44f59b97-762f-4ae0-914b-6fe370e95d72)

**Comment by Panderner** [comment]
Needs game testings after #20246

**Comment by hrydgard** [comment]
Hm, not really, since that one is for sceAudiocodec, the lower level library that sceAac uses internally. sceMp4 is a bit different, it's not just audio but also video, and it seems to be a wrapper around a lower level mp4 library called `mp4msv` to handle a lot of the parsing (but also calls sceAudiocodec for the audio).

~~Although, for sceAac specifically, it might be possible to just do DisableHLE for it and let it take care of itself, assuming the game ships a copy of sceAac.prx...~~

**Comment by hrydgard** [comment]
Actually this issue is a bit outdated. sceAac is implemented and working through an AuCtx.

The remaining issue is sceMp4, which isn't really what this issue is about. Creating a separate issue for that.
